02_using_sdram/README.rst

```diff
@@ -0,0 +1,98 @@
+Example 02: Reading and Writing SDRAM
+=====================================
+
+Most interesting SpiNNaker applications require some sort of configuration data
+to be loaded onto the machine or produce result data which must be read back
+from the machine. Typically this is done by allocating and writing/reading
+to/from the shared SDRAM on each SpiNNaker chip. In this example we'll write a
+simple SpiNNaker application which, using a single core, adds two numbers
+loaded in SDRAM and writes the answer back to SDRAM.
+
+Much of the code in this example is unchanged from the previous example so we
+will only discuss the cahnges.
+
+`Example Source code on GitHub <https://github.com/project-rig/rig_examples/tree/master/02_using_sdram>`_
+
+Allocating SDRAM
+----------------
+
+When both the host and SpiNNaker need to access the same block of SDRAM, such
+as when loading configuration data or reading back results, the SDRAM is
+typically allocated by the host.  In this example we'll allocate some SDRAM on
+chip (0, 0). We'll allocate a total of 12 bytes: 4 bytes (32 bits) for the two
+values we want to be added and another 4 bytes for the result using
+:py:meth:`~rig.machine_control.MachineController.sdram_alloc`:
+
+.. literalinclude:: adder.py
+    :language: python
+    :lines: 18
+
+The :py:meth:`~rig.machine_control.MachineController.sdram_alloc` method simply
+returns the address of a block of SDRAM on chip 0, 0 which was allocated.
+
+We also need to somehow inform the SpiNNaker application of this address. To do
+this we can use the 'tag' using the argument to give an identifier to the
+allocated memory block. The SpiNNaker application then uses the ``sark_tag_ptr`` function to look up the
+address of an SDRAM block with a specified tag.
+
+.. literalinclude:: adder.c
+    :language: c
+    :lines: 24
+
+A tag is a user-defined identifier which must be unique to the SpiNNaker chip
+and application. By convention the SDRAM allocated for applications running on
+core 1 are given tag 1, those on core 2 given tag 2 and so on. This
+conveninently means the same application binary can be loaded onto multiple
```

Comment:
Though actually I'd remove the conveniently, as it's not "convenient" it's a recommendation that is good practise.